### DISCUSSION TITLE
Multi-line labels in buttons w/ automatic wrapping

### DISCUSSION THREAD

**Discussion by sbeigel** [mantinedev/mantine#5961] [Q&A]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.6.2

### What package has an issue?

@mantine/core

### What framework do you use?

Next.js

### In which browsers you can reproduce the issue?

Other, I will specify in the bug description

### Describe the bug

As the current button label is styled with "white-space: nowrap", automatic wrapping for longer labels is not working. In "newer" browsers, I can apply "whiteSpace: 'wrap'" to make this work, but unfortunately, it doesn't work in "older" Safaris (e.g. webviews in iOS < 17.4 as the CSS text wrapping was added in 17.4).

I made my own button by copying the mantine code and by removing the "white-space: nowrap" in module.css, automatic wrapping is working in iOS < 17.4.

Is there a better way or would it be possible to make auto-wrapping configurable via a Button property?

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by sbeigel** [comment]
Ah, for older Safari versions, you must not use "wrap" but "initial" to reset/override mantine's default. So you could do:

 <Button styles={{label: {whiteSpace: 'initial'}}} ...>Multi line auto-wrapping label here...</Button>

This issue can be closed if you want to -- otherwise, I think an explicit property to control wrapping would be nice!

**Comment by rtivital** [comment]
If you want to unset certain CSS properties, you can use unset. For example, `white-space: unset`

**Comment by sbeigel** [comment]
Great, thanks, even better than "initial" (see my comment above yours). Do you think a property to control this explicitly would be "better", i.e. easier to use/discover for users? In making UIs work on desktop and mobile, I often have the use-case for multi-line/auto-wrapping buttons and mantine's default is quite restrictive: Either your buttons become really wide and break your layout, or the labels get cut...
